src/main/java/address/browser/BrowserManager.java

```diff
@@ -11,62 +12,71 @@
 import com.teamdev.jxbrowser.chromium.internal.Environment;
 
 import javafx.collections.ObservableList;
-import javafx.scene.control.TabPane;
+import javafx.scene.Node;
+import javafx.scene.layout.AnchorPane;
 
-import java.util.List;
-import java.util.Optional;
+import java.util.*;
 import java.util.logging.Level;
 
 /**
- * Manages the browser.
+ * Manages the AddressBook browser.
  */
 public class BrowserManager {
 
-
     public static final int NUMBER_OF_PRELOADED_PAGE = 3;
     public static final int PERSON_NOT_FOUND = -1;
 
-    private Optional<AddressBookBrowser> browser;
-
     private ObservableList<Person> filteredPersons;
 
-    public BrowserManager(ObservableList<Person> filteredPersons) {
+    public Optional<AddressBookPagePool> addressBookPagePool;
+
+    private AnchorPane browserPlaceHolder;
+    private Node browserDefaultScreen;
+
+    public BrowserManager(ObservableList<Person> filteredPersons, AnchorPane browserPlaceHolder,
+                          Node browserDefaultScreen) {
+        this.browserPlaceHolder = browserPlaceHolder;
+        this.browserDefaultScreen = browserDefaultScreen;
         this.filteredPersons = filteredPersons;
         String headlessProperty = System.getProperty("testfx.headless");
         if (headlessProperty != null && headlessProperty.equals("true")) {
-            browser = Optional.empty();
+            addressBookPagePool = Optional.empty();
             return;
         }
-        browser = Optional.of(new AddressBookBrowser(NUMBER_OF_PRELOADED_PAGE, this.filteredPersons));
-        browser.get().registerListeners();
         EventManager.getInstance().registerHandler(this);
+        addressBookPagePool = Optional.of(new AddressBookPagePool(NUMBER_OF_PRELOADED_PAGE));
     }
 
     @Subscribe
     public void handleLocalModelChangedEvent(LocalModelChangedEvent event){
 
-        if (!browser.isPresent()) {
+        if (!addressBookPagePool.isPresent()) {
```

Comment:
This check can be moved inside `updateBrowserContent()` so that if there is anything else that calls `updateBrowserContent()`, it won't throw error. Or put assertion in `updateBrowserContent()`.